explainer-datamodel.md

```diff
@@ -1,236 +1,329 @@
+<div style="width:150px">
+  <span id="meta_object">
+    <span>The quick </span>
+    <span style="color: brown; font-weight: bold">brown</span>
+    <span> fox jumps over the lazy dog</span>
+  </span>
+</div>
+```
+
+Another illustrative comparison shows how style on the meta object provided to `format` could be
+visualized:
+
+```js
+FormattedText.format( [ "The quick ", 
+                        { text: "brown", style: "color: brown" }, 
+                        " fox jumps over the lazy dog."
+                      ], { style: "color: blue" } );
+```
+
 ```html
 <div>
-  <span>The quick </span>
-  <span style="color: brown; font-weight: bold">brown</span>
-  <span> fox jumps over the lazy dog</span>
+  <span id="meta_object" style="color: blue">
+    <span>The quick </span>
+    <span style="color: brown">brown</span>
+    <span> fox jumps over the lazy dog</span>
+  </span>
 </div>
 ```
 
 ## CSS to achieve advanced scenarios
 
 ### Vertical Text
 
-By leveraging existing CSS styles for writing modes and other related properties, the 
-formatted text data model can support a variety of vertical text scenarios. For 
-example, by simply leveraging the `writing-mode` property on the `FormattedText` 
-object, we can orient text in its traditional vertical direction:
+By leveraging existing CSS styles for writing modes and other related properties,  
+`format` can support a variety of vertical text scenarios. For example, by simply
+leveraging the `writing-mode` property we can orient text in its traditional vertical
+direction:
 
 ```js
-let proverb = new FormattedText( "不怕慢，", "就怕站" );
-proverb.styleMap.set( "writing-mode", "vertical-rl" );
-proverb.textruns[1].styleMap.set( "font-weight", "bold" );
+let bold = "font-weight:bold";
+let meta = "writing-mode:vertical-rl;font-size:36pt";
+FormattedText.format( [ "不怕慢，", { text: "就怕站", style: bold ], meta, { height: 200 } );
 ```
 
-Together with a nice font and constrained vertical layout bounds, this would render as:
+[When rendered](explainer-rendering.md), and height-constrained as indicated, this will render as:
 
-<img src="explainerresources/vertical-text-cn.png" alt="Characters of an ancient Chineese proverb, vertically oriented in two columns, the second column bold">
+<img src="explainerresources/vertical-text-cn.png" alt="Characters of an ancient Chinese proverb, vertically oriented in two columns, the second column bold">
 
 In combination with other related CSS properties, many other vertical text layouts are
 possible:
 
 ```js
-let proverbEn = new FormattedText( "It's better to make slow progress", " than no progress at all" );
-proverbEn.styleMap.set( "writing-mode", "vertical-lr" );
-proverbEn.styleMap.set( "text-orientation", "upright" );
-proverbEn.styleMap.set( "line-height", "2" );
-proverbEn.styleMap.set( "text-align", "center" );
-proverbEn.textruns[1].styleMap.set( "font-weight", "bold" );
+let bold = "font-weight:bold";
+let styles  = "writing-mode: vertical-lr;";
```

Comment:
do we want to support property-bag syntax for the style properties, i.e.

`let styles = { writingMode: "vertical-lr", ... }`

Though I suppose hyphenated property names makes this awkward. Just thought I'd check though.

Reply:
Wait, I think this is covered by FormattedTextStyle more elegantly.

Reply:
I like the idea of property-bag styles, but as you say, hyphenated names make this awkward. The `CSSStyleDeclaration` does support these properties (legacy). It might actually be interesting to support construction of a `FormattedTextStyle` from a `CSSStyleDeclaration`. That would allow a use case like this:

```html
<span id="stylesource" style="font-weight: bold; text-align: center">some text...</span>
```
```js
FormattedText.format( "stuff to format just like the 'stylesource' element", 
                      document.querySelector("#stylesource").style
                    );
// or if the full computed styles are desired (stuff cascaded to the element)
FormattedText.format( "stuff to format just like the 'stylesource' element", 
                      self.getComputedStyle( document.querySelector("#stylesource") )
                    );
```

Reply:
I'm going to tease this idea in the WebIDL and see if anyone notices.